tests/e2e/README.md

```diff
@@ -1,4 +1,204 @@
+            ValPowers: &map[ValidatorID]uint{
+                ValidatorID("alice"): 511,
+                ValidatorID("bob"):   599,
+                ValidatorID("carol"): 500,
+            },
+        },
+    },
+},
+```
+In this sequence, we first use the `delegateTokensAction`
+to delegate 11000000 tokens from alice to herself,
+then check that alices voting power was set to 511 (note that 11000000 tokens correspond to 11 voting power here),
+then afterwards use the same action again, this time to delegate 99000000
+tokens from alice to bob, and again check the voting powers, with the change that
+bobs voting power should now be 599.
+
+For most steps, we can reuse existing code, for example
+the actions necessary to start a provider and multiple consumer chains
+are already "packaged together" and available as
+`stepsStartChains` in [steps_start_chains.go](steps_start_chains.go).
+
+**Note:** The parts of the state that are *not* defined are just *not checked*. 
+For example, if the balance of a validator is not listed in a state, it means we
+do not care about the balance of that validator in this particular state.
+
+### How to use relayers
+
+For relayers, there are currently several actions defined.
+There are some subtleties here.
+
+* `addChainToRelayer` gives the relayer the config for the chain and should be called before the relayer is expected to interact with the chain for the first time.
+* `createIbcClientsAction`, `addIbcConnectionAction` and `addIbcChannel` are used to create clients, connections and channels.
+* `relayPackets` relays all *currently queued* packets and their acknowledgements.
+* `startRelayer` starts the relayer, which means from now on, it will *automatically relay all packets*.
+Before `startRelayer`, it's possible to let packets queue up and not clear them.
+Afterwards, *all packets* will be relayed immediately, so it's no longer possible
+to e.g. delay the relaying of a packet.
+
+
+## Defining new actions
+
+It is necessary to define new actions when we want to test something that does not yet have the corresponding actions defined.
+For example, a new feature may introduce new transactions, and
+there is likely no existing action to submit these transactions to the chain.
+
+You can see the basic template for how to do this by looking at the actions in
+[actions.go](actions.go).
+The basic principle is to use `exec.Command` to execute a command inside the docker container.
+The pattern for this looks something like this:
+```
+cmd := exec.Command("docker", "exec", 
+    tr.containerConfig.InstanceName,
+    tr.chainConfigs[action.Chain].BinaryName,
+    "the command to execute, for example tx",
+    "argument 1",
+    "argument 2")
+output, err := cmd.CombinedOutput()
+if err != nil {
+    log.Fatal(err, "\n", string(bz))
+}
+
+// potentially check something in the output, or log something, ...
+```
+
+Don't forget to wire your action into [main.go](main.go):runStep, where
```

Comment:
nit: Why not point direclty to the line inside `main.go` that contains`runStep`?

Reply:
I dislike it because it will not point to the right place anymore extremely quickly, i.e. as soon as the file is touched and this is not adjusted.
I'd be happy about suggestions on how to overcome this! I know anchors exist, but I think they make the source file a bit verbose.